events/views.py

```diff
@@ -237,3 +242,96 @@ def check_participant_formset(request, event_instance, meeting_participants_form
+        self.chosen_meeting_dates = None
+        super().__init__(**kwargs)
+
+    def dispatch(self, request, meeting_id, *args, **kwargs):
+        if not request.user.is_authenticated:
+            return redirect(LOGIN_PAGE)
+
+        try:
+            EventParticipant.objects.get(user_id=request.user, event_id=meeting_id)
+
+            if self.check_user_auth(request.user, meeting_id):
+                return redirect(HOME_PAGE)
+        except Exception:
+            return redirect(HOME_PAGE)
+
+        return super(MeetingVoteView, self).dispatch(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context = {
+            'title': self.title,
+            'form': self.vote_form,
+            'event': self.chosen_event,
+            'meeting_id': self.chosen_event.id,
+            'meetings_dates': self.meetings_dates,
+            'creator': EventParticipant.objects.get(event_id=self.chosen_event, is_creator=True).user_id.username
+        }
+        return context
+
+    def get(self, request):
+        self.initialize_forms()
+        self.vote_form = self.VoteFormset()
+        return super().get(request)
+
+    def post(self, request):
+        self.initialize_forms()
+        self.vote_form = self.VoteFormset(request.POST, request.FILES)
+
+        if self.vote_form.is_valid():
+            for field, meeting in zip(self.vote_form, self.chosen_meeting_dates):
+                field_value = field.cleaned_data.get('date_vote')
+                if field_value:
+                    # add possible participant to this meeting date
+                    participant = EventParticipant.objects.get(user_id=request.user, event_id=self.chosen_event)
+                    PossibleParticipant(participant_id=participant, possible_meeting_id=meeting).save()
+
+            # send a seen request
+            notification_id = Notification.objects.filter(participant_id=participant, seen_time__isnull=True).first()
+            seen_notification(request, notification_id.id)
+            return redirect(HOME_PAGE)
+        else:
+            self.vote_form = self.VoteFormset()
+
+        return self.render_to_response(self.get_context_data())
+
+    def check_user_auth(self, user_id, meeting_id):
+        self.chosen_event = Event.objects.get(id=meeting_id)
+        self.chosen_meeting_dates = OptionalMeetingDates.objects.get_all_event_dates(self.chosen_event)
+        return PossibleParticipant.objects.did_user_vote(self.chosen_meeting_dates, user_id)  # did_user_vote
+
+    def initialize_forms(self):
+        self.title = f"{self.chosen_event.title} Vote Meeting"
+        self.VoteFormset = formset_factory(VoteForm, extra=self.chosen_meeting_dates.count())
+        self.meetings_dates = list(
+            map(lambda meeting: f"{time_format(meeting.date_time_start)} - {time_format(meeting.date_time_end)}",
+                self.chosen_meeting_dates)
+        )
```

Comment:
maybe you can use list comprehension like:
```suggestion
        self.meetings_dates = [
            f"{time_format(meeting.date_time_start)} - {time_format(meeting.date_time_end)}" for meeting in self.chosen_meeting_dates
         ]
        )
```

Reply:
Done. thanks 😄